helpers/traits/traits.php

```diff
@@ -1,239 +1,250 @@
-            '818' => 'EGP',
-            '826' => 'GBP',
-            '840' => 'USD',
-            '973' => 'AOA',
-            '974' => 'BYR',
-            '975' => 'BGN',
-            '976' => 'CDF',
-            '977' => 'BAM',
-            '978' => 'EUR',
-            '981' => 'GEL',
-            '983' => 'ECV',
-            '984' => 'BOV',
-            '986' => 'BRL',
-            '990' => 'CLF'
-        );
-        return $codes[$number];
-    }
-}
\ No newline at end of file
+trait AltapayMaster {
+
+	/**
+	 * @var bool
+	 */
+	private $api = false;
+
+	/**
+	 * Initialization of Woocommerce Payment module specifics
+	 *
+	 * @return void
+	 */
+	public function intFormFields() {
+		// Define form setting fields
+		$this->form_fields = array(
+			'enabled'        => array(
+				'title'   => __( 'Enable/Disable', 'woocommerce' ),
+				'type'    => 'checkbox',
+				'label'   => __( ' ', 'altapay' ),
+				'default' => 'yes',
+			),
+			'title'          => array(
+				'title'       => __( 'Title', 'woocommerce' ),
+				'type'        => 'text',
+				'description' => __( 'Title to show during checkout.', 'altapay' ),
+				'default'     => __( 'AltaPay', 'woocommerce' ),
+				'desc_tip'    => true,
+			),
+			'description'    => array(
+				'title'       => __( 'Message', 'woocommerce' ),
+				'description' => __( 'Message to show during checkout.', 'altapay' ),
+				'type'        => 'textarea',
+				'default'     => '',
+			),
+			'payment_action' => array(
+				'title'       => __( 'Payment action', 'woocommerce' ),
+				'description' => __( 'Make payment authorized or authorized and captured', 'altapay' ),
+				'type'        => 'select',
+				'options'     => array(
+					'authorize'         => __( 'Authorize Only', 'altapay' ),
+					'authorize_capture' => __( 'Authorize and Capture', 'altapay' ),
+				),
+				'default'     => '',
+			),
+			'currency'       => array(
+				'title'       => __( 'Currency', 'altapay' ),
+				'type'        => 'select',
+				'description' => __( 'Select the currency does this terminal work with' ),
+				'options'     => get_woocommerce_currencies(),
+				'default'     => $this->default_currency,
+			),
+		);
+	}
+
+	/**
+	 * @param int $order_id
+	 * @return array
+	 */
+	public function process_payment( $order_id ) {
+		global $woocommerce;
+		$order = new WC_Order( $order_id );
+
+		// Return goto payment url
+		return array(
+			'result'   => 'success',
+			'redirect' => $order->get_checkout_payment_url( true ),
+		);
+	}
+
+	/**
+	 * @return bool
+	 */
+	public function isAvailable() {
+		 // Check if payment method is enabled
+		if ( 'yes' !== $this->enabled ) {
+			return false;
+		}
+
+		// Check on payment currency
+		$active_currency = get_woocommerce_currency();
+
+		if ( $active_currency != $this->currency ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Tackle scenario for scheduled subscriptions
+	 *
+	 * @param float    $amount
+	 * @param WC_Order $renewal_order
+	 * @return bool|void
```

Comment:
You can't mix void with any other types. This function also behaves really oddly with only returning null and false, and also I can't see that it is used any where.